### DISCUSSION TITLE
Header propagation in istio

### DISCUSSION THREAD

**Discussion by mohite-abhi** [istio/istio#46908] [Ideas]
Is it possible to add header propagation in istio layer. So that if an app get's a header that we want it to propagate (attach the same header to any requests it makes). We might also want to make changes to such propagating headers in envoy filters.

**Comment by zirain** [comment]
try https://istio.io/latest/docs/reference/config/networking/virtual-service/#Headers

**Reply by mohite-abhi** [reply]
We have tried this, when the request arrives to a virtual service, it adds the header to the request and passes it to the destination. Now, I want to propagate this added header to whichever next service/virtualservice this destination makes a request to. Will that be possible to implement? If yes I want to try it.
